client/gutenberg/extensions/jetpack/tiled-gallery/components/tiled-gallery-item.js

```diff
@@ -0,0 +1,101 @@
+
+// Since this is a Jetpack originated block in Calypso codebase,
+// we're relaxing className rules.
+/* eslint wpcalypso/jsx-classname-namespace: 0 */
+
+/**
+ * External Dependencies
+ */
+import wp from 'wp';
+import get from 'lodash/get';
+
+/**
+ * WordPress dependencies
+ */
+const { withSelect } = wp.data;
+const { Component } = wp.element;
+
+class TiledGalleryImage extends Component {
```

Comment:
why is this a React component and not a Gutenberg element?

Reply:
@haszari has all the answers. :-)

I'd leave as much as possible refactoring the actual block for later and merge when this PR simply doesn't fatal. There's plenty to upgrade and it's still unclear if we're the ones doing it.

Reply:
Oh and it doesn't strike as an anti-pattern to me since some of the core-blocks do the same. I haven't played enough with blocks to be confident about that, tho!

Reply:
oh right - we're just copying the code. I think it's something we want to avoid unless it's mandatory to write React components. many of the core blocks do on account of being _core_ blocks and fundamental pieces. ideally blocks shouldn't stop working if the project abandons React.